tests/test_data_cleaner.py

```diff
@@ -96,3 +99,114 @@ def current_size():
+
+    def test_early_termination(self):
+        temp_folder = tempfile.mkdtemp(suffix='_test_4')
+        sub_folder_1 = tempfile.mkdtemp(dir=temp_folder)
+        sub_folder_2 = tempfile.mkdtemp(dir=temp_folder)
+        sub_folder_3 = tempfile.mkdtemp(dir=temp_folder)
+
+        current_time = time.time()
+
+        def current_size(folder):
+            try:
+                folder_size = {temp_folder: 1024*2,
+                               sub_folder_2: 200,
+                               sub_folder_3: 1400}
+                return folder_size[folder]
+            except:
+                raise ValueError("Early termination failed")
+
+        def creation_time(folder):
+            folder_time = {sub_folder_1: int(current_time - 10),
+                           sub_folder_2: int(current_time - 20),
+                           sub_folder_3: int(current_time - 30)}
+            return folder_time[folder]
+
+        with mock.patch.object(DataCleaner, 'current_size') as mocked_current_size, \
+             mock.patch.object(DataCleaner, 'creation_time') as mocked_creation_time:
+                mocked_current_size.side_effect = current_size
+                mocked_creation_time.side_effect = creation_time
+
+                # remove folders older than 1 day
+                # and remove old folders as long as total size exceed 1 MB
+                mock_cleaner = DataCleaner(temp_folder, 1024, 60 * 60 * 24)
+                try:
+                    removed = mock_cleaner.clean_up(dry_run=False)
+                except:
+                    self.fail("Early termination failed")
+                self.assertTrue(len(removed) == 1)
+                self.assertEqual(removed[0], sub_folder_3)
+                shutil.rmtree(temp_folder)
+
+    def test_filterlist(self):
+        temp_folder = tempfile.mkdtemp(suffix='_test_5')
+        _, tmp_file_1 = tempfile.mkstemp(prefix='Person1', suffix='.dcm', dir=temp_folder)
+        _, tmp_file_2 = tempfile.mkstemp(prefix='Person2', suffix='.dcm', dir=temp_folder)
+
+        filter_list = ['Person1*.dcm']
+
+        with self.assertRaises(ValueError):
+            DataCleaner(temp_folder, -1, -1, whitelist=filter_list,
+                                                  blacklist=filter_list)
+
+        mock_cleaner = DataCleaner(temp_folder, -1, -1, whitelist=filter_list)
+        removed = mock_cleaner.clean_folder(temp_folder, dry_run=True) 
+        self.assertTrue(len(removed) == 1)
+        self.assertEqual(removed[0], tmp_file_2)
+
+        mock_cleaner = DataCleaner(temp_folder, -1, -1, blacklist=filter_list)
+        removed = mock_cleaner.clean_folder(temp_folder, dry_run=True) 
+        self.assertTrue(len(removed) == 1)
+        self.assertEqual(removed[0], tmp_file_1)
+        shutil.rmtree(temp_folder)
+
+    def test_partially_remove(self):
+        """
+        Test early termination and removing the correct folders when partially removing files using a filter.
```

Comment:
If I get it right here you are testing the combination of early termination here with whitelist/blacklist. Since both things you test in other tests, does this really add necessary line coverage or can we go without it?